src/pretix/presale/views/organizer.py

```diff
@@ -97,16 +97,24 @@ def get(self, request, *args, **kwargs):
             return super().get(request, *args, **kwargs)
 
     def get_queryset(self):
-        query = Q(is_public=True) & Q(live=True)
+        query = Q(is_public=True) & Q(live=True) & Q(organizer=self.request.organizer)
+        qs = Event.objects.filter(query)
         if "old" in self.request.GET:
-            query &= Q(Q(date_from__lte=now()) & Q(date_to__lte=now()))
+            qs = qs.annotate(
+                min_to=Min('subevents__date_to')
+            ).filter(
+                Q(Q(has_subevents=False) & Q(date_from__lte=now()) & Q(date_to__lte=now())) |
```

Comment:
I like this, but I think we can work with the assumption, that `date_from` is never after `date_to`, meaning the check `date_from__lte+now()` could be removed, as `date_to` has to also be <= `now()`.